File: promo-frontend/lib/features/promotions/domain/entities/promotion.dart
import 'package:equatable/equatable.dart';

class Promotion extends Equatable {
  final int id;
  final String title;
  final String description;
  final String? imageUrl;
  final double originalPrice;
  final double discountedPrice;
  final int discountPercentage;
  final String category;
  final DateTime startDate;
  final DateTime endDate;
  final bool active;
  final String merchantName;

  const Promotion({
    required this.id,
    required this.title,
    required this.description,
    this.imageUrl,
    required this.originalPrice,
    required this.discountedPrice,
    required this.discountPercentage,
    required this.category,
    required this.startDate,
    required this.endDate,
    required this.active,
    required this.merchantName,
  });

  @override
  List<Object?> get props => [
        id,
        title,
        description,
        imageUrl,
        originalPrice,
        discountedPrice,
        discountPercentage,
        category,
        startDate,
        endDate,
        active,
        merchantName,
      ];
}

File: promo-backend/promo-domain/src/main/java/com/promo/domain/model/Promotion.java
package com.promo.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Promotion {
    private Long id;
    private String title;
    private String description;
    private String imageUrl;
    private BigDecimal originalPrice;
    private BigDecimal discountedPrice;
    private Integer discountPercentage;
    private String category;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private Boolean active;
    private String merchantName;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
